pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTaskRestletResource.java

```diff
@@ -0,0 +1,387 @@
+   *     Get the task config for the specified task name.
+   *   </li>
+   *   <li>
+   *     "/tasks/taskstates/{taskType}":
+   *     Get a map from task name to task state for the specified task type.
+   *   </li>
+   *   <li>
+   *     "/tasks/taskstate/{taskName}":
+   *     Get the task state for the specified task name.
+   *   </li>
+   *   <li>
+   *     "/tasks/taskqueues":
+   *     List all task queues.
+   *   </li>
+   *   <li>
+   *     "/tasks/taskqueuestate/{taskType}":
+   *     Get the task queue state for the specified task type.
+   *   </li>
+   * </ul>
+   */
+  @Get
+  @Override
+  public Representation get() {
+    ConcurrentMap<String, Object> attributes = getRequest().getAttributes();
+    String taskName = (String) attributes.get(TASK_NAME);
+    String taskType = (String) attributes.get(TASK_TYPE);
+
+    String requestType = getReference().getSegments().get(1);
+    switch (requestType) {
+      case "tasktypes":
+        // "/tasks/tasktypes"
+        return getTaskTypes();
+      case "tasks":
+        // "/tasks/tasks/{taskType}"
+        return getTasks(taskType);
+      case "taskconfig":
+        // "/tasks/taskconfig/{taskName}"
+        return getTaskConfig(taskName);
+      case "taskstates":
+        // "/tasks/taskstates/{taskType}"
+        return getTaskStates(taskType);
+      case "taskstate":
+        // "/tasks/taskstate/{taskName}"
+        return getTaskState(taskName);
+      case "taskqueues":
+        // "/tasks/taskqueues"
+        return getTaskQueues();
+      case "taskqueuestate":
+        // "/tasks/taskqueuestate/{taskType}"
+        return getTaskQueueState(taskType);
+      default:
+        throw new IllegalStateException("Unsupported request type: " + requestType);
+    }
+  }
+
+  /**
+   * URI Mappings:
+   * <ul>
+   *   <li>
+   *     "/tasks/taskqueue/{taskType}":
+   *     Create a task queue for the specified task type.
+   *   </li>
+   *   <li>
+   *     "/tasks/task/{taskType}":
+   *     Submit a task of the specified task type to the task queue.
+   *   </li>
+   * </ul>
+   */
+  @Post
+  @Override
+  public Representation post(Representation entity) {
+    String taskType = (String) getRequest().getAttributes().get(TASK_TYPE);
+    String requestType = getReference().getSegments().get(1);
+    switch (requestType) {
+      case "taskqueue":
+        // "/tasks/taskqueue/{taskType}"
+        return createTaskQueue(taskType);
+      case "task":
+        // "/tasks/task/{taskType}"
+        return submitTask(taskType, entity);
+      default:
+        throw new IllegalStateException("Unsupported request type: " + requestType);
+    }
+  }
+
+  /**
+   * URI Mappings:
+   * <ul>
+   *   <li>
+   *     "/tasks/taskqueue/{taskType}?state={state}":
+   *     Stop/resume a task queue based on the specified {state} (stop|resume).
+   *   </li>
+   *   <li>
+   *     "/tasks/scheduletasks":
+   *     Schedule tasks.
+   *   </li>
+   * </ul>
+   */
+  @Put
```

Comment:
According to http://stackoverflow.com/questions/630453/put-vs-post-in-rest, POST is used to create, PUT is used to create or update